pkg/mutators/podtemplatespec.go

```diff
@@ -25,63 +29,82 @@ func NewPodTemplateSpec(opts ...OptionsFunc) *PodTemplateSpec {
 	return pts
 }
 
-func (p *PodTemplateSpec) Mutate(inputPts corev1.PodTemplateSpec, limitRangeMemory *limitrange.Config) (corev1.PodTemplateSpec, error) {
+func (p *PodTemplateSpec) Mutate(ctx context.Context, inputPts corev1.PodTemplateSpec, limitRangeMemory *limitrange.Config) (corev1.PodTemplateSpec, error) {
 
 	pts := *inputPts.DeepCopy()
 	if limitRangeMemory == nil {
-		return pts, fmt.Errorf("invalid limit range config")
+		return pts, p.errorIfNotDryRun(ctx, "invalid limit range config")
 	}
 
-	if err := p.setAndValidateResourceRequirements(pts.Spec.InitContainers, limitRangeMemory); err != nil {
+	if err := p.setAndValidateResourceRequirements(ctx, pts.Spec.InitContainers, limitRangeMemory); err != nil {
 		return pts, err
 	}
 
-	if err := p.setAndValidateResourceRequirements(pts.Spec.Containers, limitRangeMemory); err != nil {
+	if err := p.setAndValidateResourceRequirements(ctx, pts.Spec.Containers, limitRangeMemory); err != nil {
 		return pts, err
 	}
 
 	return pts, nil
 }
 
-func (p *PodTemplateSpec) setAndValidateResourceRequirements(containers []corev1.Container, limitRangeMemory *limitrange.Config) error {
+func (p *PodTemplateSpec) setAndValidateResourceRequirements(ctx context.Context, containers []corev1.Container, limitRangeMemory *limitrange.Config) error {
 	for idx := range containers {
 		container := &containers[idx]
+		if p.dryRun {
+			// On dry-run use a copy to go through the motions, do not modify original
+			container = container.DeepCopy()
+		}
 
-		p.setMemoryRequest(container, limitRangeMemory)
-		p.setMemoryLimit(container, limitRangeMemory)
+		p.setMemoryRequest(ctx, container, limitRangeMemory)
+		p.setMemoryLimit(ctx, container, limitRangeMemory)
 
-		if err := p.validateMemoryRequirements(*container, limitRangeMemory); err != nil {
+		if err := p.validateMemoryRequirements(ctx, *container, limitRangeMemory); err != nil {
 			return err
 		}
```

Comment:
We need to allow the request on dry-run still. In theory this should be valid since we set everything up but we want to allow all requests even if they would be rejected by limitranger anyway.

Reply:
Ok, fixed